src/ansys/dynamicreporting/core/utils/report_objects.py

```diff
@@ -3473,49 +3473,129 @@ class statisticalREST(GeneratorREST):
     """
     Representation of a statistical analysis generator.
 
-    Defines getters and setters for analysis type, predictor variables, response
-    variables, and intercept.
+    Defines getters and setters for analysis type, table name, tree name, predictor
+    variables, response variables, and analysis_parameters. For variables the standard
+    input format is 2D array to support multiple variables.
     """
 
     def __init__(self):
         super().__init__()
 
-    @property
-    def analysis_type(self) -> str:
-        return self.get_property().get("analysis_type")
+    def get_analysis_type(self):
+        if "stats_params" in json.loads(self.params):
+            if "analysis_type" in json.loads(self.params)["stats_params"]:
+                return json.loads(self.params)["stats_params"]["analysis_type"]
+        return ""
 
-    @analysis_type.setter
-    def analysis_type(self, value: str) -> None:
-        props = self.get_property()
-        props["analysis_type"] = value
-        self.set_property(props)
+    def set_analysis_type(self, value=""):
+        if type(value) is not str:
+            raise ValueError("Error: analysis type should be a string")
+        # supported analysis types array to be expanded
+        analysis_types = ["Linear Regression"]
+        if value not in analysis_types:
+            raise ValueError(
+                "Error: input does not belong to the set of supported analysis types. Must be one of [{a_types}].".format(
+                    a_types=", ".join(analysis_types)
+                )
+            )
+        d = json.loads(self.params)
+        if "stats_params" not in d:
+            d["stats_params"] = {}
+        d["stats_params"]["analysis_type"] = value
+        self.params = json.dumps(d)
+        return
```

Comment:
The `return` is implied and can be removed.

Reply:
Gotcha, thank you Visesh, didn't consider how this is implied before